### ISSUE TITLE
Can't decode self-described documents under 0.11

### ISSUE THREAD

**Issue by vorner** [pyfisch/cbor#177]
Hello

Let's say I have this short snippet of code:

```rust
use std::collections::HashMap;

use serde::Serialize;
use serde_cbor;
use serde_cbor::ser::Serializer;

type TestData = HashMap<String, usize>;

fn main() {
    let mut d = TestData::new();
    d.insert("Hello".to_owned(), 42);
    let mut output = Vec::new();
    let mut serializer = Serializer::new(&mut output);
    serializer.self_describe().unwrap();
    d.serialize(&mut serializer).unwrap();
    let d: HashMap<String, usize> = serde_cbor::from_slice(&output[..]).unwrap();
    println!("{:?}", d);
}
```

If I compile it with serde_cbor 0.10, everything is fine. If I switch to 0.11, it panics with this error:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ErrorImpl { code: Message("invalid type: newtype struct, expected a map"), offset: 0 }', src/libcore/result.rs:1192:5
```

It's panicking on the `serde_cbor::from_slice` line. If I remove the `serializer::self_describe` line, everything is fine again.

Is this a regression or should I be checking for the self-description magic number somehow manually?

**Comment by pyfisch** [comment]
Have you enabled the `tags` feature?

**Comment by vorner** [comment]
Originally no, but even now when I add it, the problem still stays.

**Comment by pyfisch** [comment]
Huh, that's weird. cc @rklaehn

**Comment by vorner** [comment]
Let me attach the Cargo.{toml,lock} so we can be sure to be talking about the same code.

**Comment by pyfisch** [comment]
I just released v0.11.1 fixing it. If you disable the tags feature in the serde-cbor-reproducer it does work with 0.11.1.

If you enable tags you must handle them in any place they might occur.

**Comment by vorner** [comment]
Thank you, I can confirm it fixes the problem for me.
